ecl/hql/hqllex.l

```diff
@@ -64,6 +64,94 @@ extern void hex2str(char * target, const char * digits, unsigned len);
 #define RETURNHARD(sym) \
         setupdatepos; return sym;
 
+//#define TEST_ID_SUGGESTIONS
+#ifdef TEST_ID_SUGGESTIONS
+static char getNewIdChar(unsigned index)
+{
+    //1st char of an ID must be an alpha [a-z]
+    if (index == 0)
+        return 'a' + rand()%26;
+
+    unsigned n = rand()%37;
+    if (n < 10)
+        return '0' + n;
+    else if(n == 10)
+        return '_';
+    else if (n > 10  && n < 37)
+        return 'a' + n - 11;
+    else
+        throwUnexpected();
+}
+
+static unsigned scrambledIdCount = 0;
+
+class DummyAtom : public CInterfaceOf<IIdAtom>
```

Comment:
This is almost certainly not wanted.  You should create an IdAtom in the same way as normal.